index.js

```diff
@@ -28,6 +29,15 @@ async function validateTemplateStructure(templateObj) {
     return data;
 }
 
+// eslint-disable-next-line require-jsdoc
+async function validatePipelineTemplateStructure(templateObj) {
+    const data = await SCHEMA_PIPELINE_CONFIG.validateAsync(templateObj, {
+        abortEarly: false
+    });
+
+    return data;
+}
+
```

Comment:
Per discussion, refactor existing function `validateTemplateStructure` to accept the schema as one of the argument
```suggestion
```